### PR TITLE
add `load_from_std_lib` function to `Lua` state

### PR DESCRIPTION
**Pull request by azdle**
This function allows loading of any portion of the standard libraries
after initial state creation.

Specifically, this is filling a need to load the debug library after an
error is encountered so that a traceback can be generated without
risking exposing the debug library anywhere else in the code.

### CODE REVIEW

**Comment by kyren** [comment]
Sorry for sitting on this for so long...

I've been trying to figure out exactly what I'm going to do re: #116 for a while, and this PR sort of overlaps with it.

I still need to figure out what I'm going to do, and that will affect everything that loads the Lua stdlib, but I don't think that's a valid reason to wait on this PR, since this is still a strict improvement.

Would it be possible to just combine the body here with the code that loads libraries in `create_lua`?  I'd be happy to merge this then.

**Comment by azdle** [comment]
No worries.

Not sure why I didn't do this in the first place. Force-pushed with the big block of `if`s moved out to a common function.

Also, CI seems to be failing on some not totally backwards compatible change to rust in 1.35 -> 1.36 that's causing rustyline to fail to build. `s/3.0.0/5.0.0/` for rustyline fixes it. Do you want me to just push that in this PR so it builds? Or do you want that change done somewhere else to keep things clean?

**Comment by kyren** [comment]
You can go ahead and push that as part of this PR if you don't mind, I don't mind the drive-by change honestly.

**Comment by azdle** [comment]
There was also a test case failing due to some changes in what errors the compiler produces for broken code. Updated a couple of tests in `tests/compile-fail` to what 1.36 uses too. Hopefully that's right, it looks like only the current stable is built in CI so that should get the shiny green checkmark here.

File: src/lua.rs [hunk 1]
```diff
@@ -513,4 +485,48 @@ unsafe fn create_lua(lua_mod_to_load: StdLib) -> Lua {
     }
 }
 
+unsafe fn load_from_std_lib(state: *mut ffi::lua_State, lua_mod: StdLib) {
+    // Do not open the debug library, it can be used to cause unsafety.
```
Comment (kyren) [inline src/lua.rs]: I know you didn't add this, but would you mind removing this comment?

File: src/lua.rs [hunk 2]
```diff
@@ -125,6 +125,18 @@ impl Lua {
         create_lua(lua_mod)
     }
 
+    /// Loads standard the specified set of the standard libraries into an existing Lua state.
+    ///
+    /// Use the [`StdLib`] flags to specifiy the libraries you want to load.
+    ///
+    /// This function is unsafe because it can be used to load the `debug` library which can be used
+    /// to break the safety guarantees provided by rlua.
+    pub unsafe fn load_from_std_lib(&self, lua_mod: StdLib) -> Result<()> {
```
Comment (kyren) [inline src/lua.rs]: Could this be `load_std_lib` and also can there be an `unsafe_load_std_lib` variant that allows debug?

Comment (azdle) [reply src/lua.rs]: `load_std_lib` implies to me that it loads the whole std lib. Maybe `load_from_std_lib` & `unsafe_load_from_std_lib`? Also any preference on if the first should error if debug is in the list or just ignore it?

Comment (kyren) [reply src/lua.rs]: Yeah you're right, I see your point on `load_from_std_lib`, that's fine.

The safe version should *panic* I think if it's asked to load the debug library (that's what new does).

Comment (azdle) [reply src/lua.rs]: Ah, should have looked at that. :+1: 

Changed to `load_from_std_lib` and `unsafe_load_from_std_lib`. (Also squashed the fixups.)

**Comment by azdle** [comment]
Had to force-push again because I forgot to `cargo fmt` and I had an extra blank line.

**Comment by kyren** [comment]
Awesome work, thank you very much!

**Comment by azdle** [comment]
Thank *you* for this wonderful library!

**Pull request merged**